src/rhsm_debug/debug_commands.py

```diff
@@ -174,6 +174,9 @@ def _do_command(self):
 
                 # try to rename the dir atomically
                 # rename only works on the same filesystem, but it is atomic.
+
+                if os.stat(content_path).st_dev != os.stat(dest_dir_name).st_dev:
+                    raise InvalidCLIOptionError(_("To use the no-archive option, the destination directory '%s' must exist on the same file system as the data assembly directory '%s'." % (dest_dir_name, content_path)))
```

Comment:
For this case, we can probably assume the parent dir of dest_dir_name is on the same file system[1] as dest_dir_name (and ditto content_path), so we could compare assemble_path and self.options.destination, which we have earlier

(see https://github.com/candlepin/subscription-manager/commit/1504b4c25f54c9c1644672e60b50e08bd5d7e530 for a totally untested version)

[1] in theory, dest dir could be say, /mnt/my_rhsm_debug_path, and /mnt/my_rhsm_debug_path could be a mount point for a different fs, but thats unlikely.